Tasks/PowerShellOnTargetMachines2/PowerShellOnTargetMachines.ps1

```diff
@@ -0,0 +1,43 @@
+Trace-VstsEnteringInvocation $MyInvocation
+$global:ErrorActionPreference = 'Continue'
+
+# Undocumented VstsTaskSdk variable so Verbose/Debug isn't converted to ##vso[task.debug].
+# Otherwise any content the ad-hoc script writes to the verbose pipeline gets dropped by
+# the agent when System.Debug is not set.
+$global:__vstsNoOverrideVerbose = $true
+
+try {
+    Import-VstsLocStrings "$PSScriptRoot\task.json"
+    
+    . "$PSScriptRoot\Utility.ps1"
+    . "$PSScriptRoot\SessionHelper.ps1"
+    . "$PSScriptRoot\JobHelper.ps1"
+    . "$PSScriptRoot\RunPowerShellScriptJob.ps1"
+
+    # Get all inputs for the task
+    $input_Machines = Get-VstsInput -Name "Machines" -Require -ErrorAction "Stop"
+    $input_UserName = Get-VstsInput -Name "UserName" -Require -ErrorAction "Stop"
+    $input_UserPassword = ConvertTo-SecureString -AsPlainText -String $(Get-VstsInput -Name "UserPassword" -Require -ErrorAction "Stop") -Force
+
+    $input_Protocol = Get-VstsInput -Name "Protocol" -Require -ErrorAction "Stop"
+    $input_NewPsSessionOptionArguments = Get-VstsInput -Name "NewPsSessionOptionArguments"
+
+    $input_RunPowershellInParallel = Get-VstsInput -Name "RunPowershellInParallel" -AsBool
+
+    $targetMachineNames = Parse-TargetMachineNames -machineNames $input_Machines
+    $credential = Get-TargetMachineCredential -userName $input_UserName -securePassword $input_UserPassword -variableNames "input_UserName","input_UserPassword"
+
+    $PSSessionOption = Get-NewPSSessionOption -arguments $input_NewPsSessionOptionArguments
+
+    $sessions = Create-PSSessionToRemoteMachines -targetMachineNames $targetMachineNames -targetMachineCredential $credential -protocol $input_Protocol
```

Comment:
How is the error reporting when some of the machines fail. Can you share the experience

Reply:
refer [this](https://github.com/Microsoft/vsts-tasks/pull/6828/files#diff-55655aeb587cea1038f9ec106c3f7f32L1) for how we handle pssession connection establishment errors